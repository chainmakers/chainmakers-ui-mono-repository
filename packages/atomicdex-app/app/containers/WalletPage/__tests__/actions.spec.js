// @flow
import {
  loadTransactions,
  loadTransactionsSuccess,
  openWithdrawModal,
  closeWithdrawModal,
  openDepositModal,
  closeDepositModal,
  loadCoinTransactions,
  openJoyride,
  closeJoyride,
  loadWithdraw,
  loadWithdrawSuccess,
  loadWithdrawError
} from '../actions';
import {
  TRANSACTIONS_LOAD,
  LOAD_TRANSACTIONS_SUCCESS,
  WITHDRAW_MODAL_OPEN,
  WITHDRAW_MODAL_CLOSE,
  DEPOSIT_MODAL_OPEN,
  DEPOSIT_MODAL_CLOSE,
  COIN_TRANSACTIONS_LOAD,
  JOYRIDE_OPEN,
  JOYRIDE_CLOSE,
  WITHDRAW_LOAD,
  WITHDRAW_LOAD_SUCCESS,
  WITHDRAW_LOAD_ERROR
} from '../constants';
import type { ErrorType } from '../../schema';

describe('containers/WalletPage/actions/loadTransactions', () => {
  it('should loadTransactions should create loadTransactions action', () => {
    expect(loadTransactions()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: TRANSACTIONS_LOAD
    };

    expect(loadTransactions()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/loadTransactionsSuccess', () => {
  it('should loadTransactionsSuccess should create loadTransactionsSuccess action', () => {
    expect(loadTransactionsSuccess()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_TRANSACTIONS_SUCCESS
    };

    expect(loadTransactionsSuccess()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/openWithdrawModal', () => {
  const coin = 'KMD';
  it('should loadTransactionsSuccess should create openWithdrawModal action', () => {
    expect(openWithdrawModal(coin)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: WITHDRAW_MODAL_OPEN,
      payload: {
        coin
      }
    };

    expect(openWithdrawModal(coin)).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/closeWithdrawModal', () => {
  it('should loadTransactionsSuccess should create closeWithdrawModal action', () => {
    expect(closeWithdrawModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: WITHDRAW_MODAL_CLOSE
    };

    expect(closeWithdrawModal()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/openDepositModal', () => {
  const coin = 'KMD';
  it('should loadTransactionsSuccess should create openDepositModal action', () => {
    expect(openDepositModal(coin)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: DEPOSIT_MODAL_OPEN,
      payload: {
        coin
      }
    };

    expect(openDepositModal(coin)).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/closeDepositModal', () => {
  it('should loadTransactionsSuccess should create closeDepositModal action', () => {
    expect(closeDepositModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: DEPOSIT_MODAL_CLOSE
    };

    expect(closeDepositModal()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/loadCoinTransactions', () => {
  const data = {
    result: 'success',
    status: 'queued',
    coin: 'KMD',
    queueId: 23
  };
  it('should loadCoinTransactions should create loadCoinTransactions action', () => {
    expect(loadCoinTransactions(data)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: COIN_TRANSACTIONS_LOAD,
      payload: data
    };

    expect(loadCoinTransactions(data)).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/openJoyride', () => {
  it('should openJoyride should create openJoyride action', () => {
    expect(openJoyride()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: JOYRIDE_OPEN
    };

    expect(openJoyride()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/closeJoyride', () => {
  it('should closeJoyride should create closeJoyride action', () => {
    expect(closeJoyride()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: JOYRIDE_CLOSE
    };

    expect(closeJoyride()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/loadWithdraw', () => {
  const payload = {
    amount: 0.1,
    address: 'RRVJBpA5MoeTo3beA1iP6euWWrWcJdJtXu',
    coin: 'BEER'
  };
  it('should loadWithdraw should create loadWithdraw action', () => {
    expect(loadWithdraw(payload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: WITHDRAW_LOAD,
      payload
    };

    expect(loadWithdraw(payload)).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/loadWithdrawSuccess', () => {
  const payload = {
    amount: 0.1,
    address: 'RRVJBpA5MoeTo3beA1iP6euWWrWcJdJtXu',
    coin: 'BEER'
  };
  it('should loadWithdrawSuccess should create loadWithdrawSuccess action', () => {
    expect(loadWithdrawSuccess(payload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: WITHDRAW_LOAD_SUCCESS,
      payload
    };

    expect(loadWithdrawSuccess(payload)).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/loadWithdrawError', () => {
  const error: ErrorType = {
    context: {
      action: WITHDRAW_LOAD,
      params: {
        amount: 0.1,
        address: 'RRVJBpA5MoeTo3beA1iP6euWWrWcJdJtXu',
        coin: 'BEER'
      }
    },
    type: 'RPC',
    message: "can't connect to electrum server"
  };

  it('should loadWithdrawError should create loadWithdrawError action', () => {
    expect(loadWithdrawError(error)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: WITHDRAW_LOAD_ERROR,
      error
    };

    expect(loadWithdrawError(error)).toEqual(expectedResult);
  });
});
