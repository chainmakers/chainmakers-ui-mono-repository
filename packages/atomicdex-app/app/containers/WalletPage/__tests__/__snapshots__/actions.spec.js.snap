// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/WalletPage/actions/closeAssetModal should loadTransactionsSuccess should create closeAssetModal action 1`] = `
Object {
  "type": "atomicapp/WalletPage/ASSET_MODAL_CLOSE",
}
`;

exports[`containers/WalletPage/actions/closeJoyride should closeJoyride should create closeJoyride action 1`] = `
Object {
  "type": "atomicapp/WalletPage/JOYRIDE_CLOSE",
}
`;

exports[`containers/WalletPage/actions/closeRemoveElectrumModal should loadTransactionsSuccess should create closeRemoveElectrumModal action 1`] = `
Object {
  "type": "atomicapp/WalletPage/REMOVING_ELECTRUM_MODAL_CLOSE",
}
`;

exports[`containers/WalletPage/actions/loadCoinTransactions should loadCoinTransactions should create loadCoinTransactions action 1`] = `
Object {
  "payload": Object {
    "coin": "KMD",
    "queueId": 23,
    "result": "success",
    "status": "queued",
  },
  "type": "atomicapp/WalletPage/COIN_TRANSACTIONS_LOAD",
}
`;

exports[`containers/WalletPage/actions/loadTransactions should loadTransactions should create loadTransactions action 1`] = `
Object {
  "type": "atomicapp/WalletPage/TRANSACTIONS_LOAD",
}
`;

exports[`containers/WalletPage/actions/loadTransactionsSuccess should loadTransactionsSuccess should create loadTransactionsSuccess action 1`] = `
Object {
  "type": "atomicapp/WalletPage/LOAD_TRANSACTIONS_SUCCESS",
}
`;

exports[`containers/WalletPage/actions/loadWithdraw should loadWithdraw should create loadWithdraw action 1`] = `
Object {
  "payload": Object {
    "address": "RRVJBpA5MoeTo3beA1iP6euWWrWcJdJtXu",
    "amount": 0.1,
    "coin": "BEER",
  },
  "type": "atomicapp/WalletPage/WITHDRAW_LOAD",
}
`;

exports[`containers/WalletPage/actions/loadWithdrawError should loadWithdrawError should create loadWithdrawError action 1`] = `
Object {
  "error": Object {
    "context": Object {
      "action": "atomicapp/WalletPage/WITHDRAW_LOAD",
      "params": Object {
        "address": "RRVJBpA5MoeTo3beA1iP6euWWrWcJdJtXu",
        "amount": 0.1,
        "coin": "BEER",
      },
    },
    "message": "can't connect to electrum server",
    "type": "RPC",
  },
  "type": "atomicapp/WalletPage/WITHDRAW_LOAD_ERROR",
}
`;

exports[`containers/WalletPage/actions/loadWithdrawSuccess should loadWithdrawSuccess should create loadWithdrawSuccess action 1`] = `
Object {
  "payload": Object {
    "address": "RRVJBpA5MoeTo3beA1iP6euWWrWcJdJtXu",
    "amount": 0.1,
    "coin": "BEER",
  },
  "type": "atomicapp/WalletPage/WITHDRAW_LOAD_SUCCESS",
}
`;

exports[`containers/WalletPage/actions/openAssetModal should loadTransactionsSuccess should create openAssetModal action 1`] = `
Object {
  "payload": Object {
    "coin": "KMD",
    "tab": 1,
  },
  "type": "atomicapp/WalletPage/ASSET_MODAL_OPEN",
}
`;

exports[`containers/WalletPage/actions/openJoyride should openJoyride should create openJoyride action 1`] = `
Object {
  "type": "atomicapp/WalletPage/JOYRIDE_OPEN",
}
`;

exports[`containers/WalletPage/actions/openRemoveElectrumModal should loadTransactionsSuccess should create openRemoveElectrumModal action 1`] = `
Object {
  "payload": Object {
    "coin": "KMD",
  },
  "type": "atomicapp/WalletPage/REMOVING_ELECTRUM_MODAL_OPEN",
}
`;

exports[`containers/WalletPage/actions/switchTabAssetInfo should switchTabAssetInfo should create switchTabAssetInfo action 1`] = `
Object {
  "payload": Object {
    "tab": 2,
  },
  "type": "atomicapp/DexPage/TAB_ASSET_INFO_SWITCH",
}
`;
